app/controllers/repair_requests_controller.rb

```diff
@@ -1,57 +1,39 @@
 class RepairRequestsController < ApplicationController
   before_action :set_property
+  before_action :set_cautionary_contact
+  before_action :create_and_set_keyfax_session
+  before_action :set_keyfax_result
 
   def new
-
-    if keyfax_session_started?
-      if should_get_keyfax_result?
-        @keyfax_result = Hackney::KeyfaxResult.find(params["guid"])
-        unless sor_code_valid?
-          flash.now[:error] = [@keyfax_result.problem_description]
-        end
-      else
-        flash.now[:error] = ["Keyfax session cancelled"]
-      end
-    end
-
-    @cautionary_contact = Hackney::CautionaryContact.find_by_property_reference(@property.reference)
-    @keyfax_session = Hackney::KeyfaxSession.create(current_page_url: new_property_repair_request_url(@property.reference))
-    cautionary_contact_alert_codes = @cautionary_contact.alert_codes.join("; ")
-
-    @repair_request = Hackney::RepairRequest.new(
-      contact_attributes: {},
-      work_orders_attributes: [{
-        sor_code: @keyfax_result&.sor_code,
-        }],
-      priority: @keyfax_result&.priority,
-      description: "#{cautionary_contact_alert_codes} #{@keyfax_result&.fault_text}"
+    @repair_request = new_repair_request(
+      contact_attributes: {
```

Comment:
Do you need this `contact_attributes`?

Reply:
Yes, otherwise `@repair_request.contact` will be nil and the fields won't show up in the form.

Reply:
If it is a dependency then probably this should be into the constructor of the `Hackney::RepairRequest` or in [def new_repair_request](https://github.com/LBHackney-IT/repairs-management/pull/248/files/511e347b977f44893373f549e4334c2884ec4c32#diff-9210ff119b95d40791d2f4a1b689d3baR93)

Reply:
I see this is more like a default value than a dependency. The model should still work with a nil `contact`.

Reply:
And I only want it to be set to this on `new`, but not on `create`.